components/tidb_query/src/expr/builtin_like.rs

```diff
@@ -13,19 +13,11 @@ impl ScalarFunc {
         let target = try_opt!(self.children[0].eval_string(ctx, row));
         let pattern = try_opt!(self.children[1].eval_string(ctx, row));
         let escape = try_opt!(self.children[2].eval_int(ctx, row)) as u32;
-        Ok(Some(
-            (match self.field_type.as_accessor().collation() {
```

Comment:
Happen to see that quite a lot of `as_accessor` are actually useless?

Reply:
yes....... I am wondering is there any way to completely get rid of them.